src/qat/purr/backends/qblox/codegen.py

```diff
@@ -331,34 +346,41 @@ def _process_waveform(self, waveform: Waveform, target: PulseChannel):
         pulse *= scale
         pulse += target.bias
 
-        if 2 * pulse.size > self._wf_memory:  # for both I and Q
+        if ((pulse.real < -1) | (pulse.real > 1)).any():
```

Comment:
do we not need this to be the magnitude of the pulse? If not can we add a comment to be explicit that we're testing components, not magnitude here